File: test_data/code/python/1863-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        memo = {}
        def dfs(mask):
            if mask in memo:
                return memo[mask]
            res = 0
            for i, n in enumerate(nums):
                if mask & (1 << i):
                    res += dfs(mask ^ (1 << i)) ^ n
                else:
                    res += dfs(mask | (1 << i))
            memo[mask] = res
            return res
        return dfs(0)

File: test_data/code/java/1863-sum-of-all-subset-xor-totals.java
class Solution {
    public int subsetXORSum(int[] nums) {
        int res = 0;
        for (int mask = 0; mask < (1 << nums.length); mask++) {
            int xor = 0;
            for (int i = 0; i < nums.length; i++) {
                if (((mask >> i) & 1) == 1) {
                    xor ^= nums[i];
                }
            }
            res += xor;
        }
        return res;
    }
}